olive/cli/capture_onnx.py

```diff
@@ -0,0 +1,228 @@
+            default=False,
+            required=False,
+            help="Remove language modeling head from your ONNX model.",
+        )
+        mb_group.add_argument(
+            "--enable_cuda_graph",
+            type=bool,
+            default=None,  # Explicitly setting to None to differentiate between user intent and default.
+            required=False,
+            help=(
+                "The model can use CUDA graph capture for CUDA execution provider. "
+                "If enabled, all nodes being placed on the CUDA EP is the prerequisite "
+                "for the CUDA graph to be used correctly."
+            ),
+        )
+
+        # remote options
+        add_remote_options(sub_parser)
+
+        sub_parser.set_defaults(func=CaptureOnnxGraphCommand)
+
+    def run(self):
+        from olive.workflows import run as olive_run
+
+        set_tempdir(self.args.output_path)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            run_config = self.get_run_config(tempdir)
+
+            olive_run(run_config)
+
+            if is_remote_run(self.args):
+                # TODO(jambayk): point user to datastore with outputs or download outputs
+                # both are not implemented yet
+                return
+
+            output_path = Path(self.args.output_path)
+            logger.info("Model and adapters are saved to %s", output_path.resolve())
+
+    def get_run_config(self, tempdir: str) -> Dict:
+        config = deepcopy(TEMPLATE)
+
+        config["input_model"]["model_path"] = get_model_name_or_path(self.args.model_name_or_path)
+        if self.args.task is not None:
+            config["input_model"]["task"] = self.args.task
+
+        config["output_dir"] = self.args.output_path
+
+        if self.args.use_model_builder:
+            del config["passes"]["c"]
+            config["passes"]["m"]["precision"] = self.args.precision
+            if self.args.int4_block_size is not None:
+                config["passes"]["m"]["int4_block_size"] = self.args.int4_block_size
+            if self.args.int4_accuracy_level is not None:
+                config["passes"]["m"]["int4_accuracy_level"] = self.args.int4_accuracy_level
+            config["passes"]["m"]["exclude_embeds"] = self.args.exclude_embeds
+            config["passes"]["m"]["exclude_lm_head"] = self.args.exclude_lm_head
+            config["passes"]["m"]["enable_cuda_graph"] = self.args.enable_cuda_graph
+        else:
+            del config["passes"]["m"]
+            config["passes"]["c"]["past_key_value_name"] = self.args.past_key_value_name
+            config["passes"]["c"]["device"] = self.args.device
+            config["passes"]["c"]["torch_dtype"] = self.args.torch_dtype
+            config["passes"]["c"]["target_opset"] = self.args.target_opset
+            config["passes"]["c"]["use_dynamo_exporter"] = self.args.use_dynamo_exporter
+            config["passes"]["c"]["save_metadata_for_token_generation"] = self.args.use_ort_genai
```

Comment:
This should also move to some central implementation so more than one such cli command use it and not have to author/change in multiple places.